rfc/rfc019-employee-identity-means.md

```diff
@@ -0,0 +1,187 @@
+New feature or applications require a higher assurance level for authentication than the current situation.
+This introduces an additional authentication step in the process of users. This often happens at a place in the workflow where they don't expect it.
+This is a situation that remain until the main login is adapted to the newer/higher security standard.
+This RFC bridges the gap so the software can already be changed to accept newer protocols and the user can be made aware of upcoming changes without going through an authentication step using additional means.
+
+## 2. Terminology
+
+* **Client application**: The application that requires access.
+* **Node**: A piece of software implementing the Nuts RFCs.
+* **Resource server**: The application \(a protected resource\) that requires authorized access to its API’s.
+* **VerifiableCredential**: Verifiable Credential according to the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+* **VerifiablePresentation**: Verifiable Presentation according to the [Verifiable Credential Data Model](https://www.w3.org/TR/vc-data-model/).
+
+## 3. Nuts Employee Identity
+
+The authentication flow is as follows:
+
+- a user requests access to a resource that is hosted at a *resource server* and requires additional authentication.
+- the *client application* starts a session on the *node* with the current user data.
+- the user is directed to a web page where it accepts the given conditions by pressing a button.
+- this button initiates a POST request to the *node for the given session.
+- upon receiving the request, the node will issue a `NutsEmployeeCredential` from and to the organization DID of the current user context.
+- the credential is used to create a VerifiablePresentation.
+- the VerifiablePresentation is used in the access token request as defined in [RFC003](./rfc003-oauth2-authorization.md#422-payload).
+
+### 3.1 User data
+
+In order for the resource server to comply to local legislation the following data MUST be added to the credential:
+
+- **Initials**: The initials of the user.
+- **Familyname**: The family name of the user.
+- **Identifier**: A unique identifier for the user within the context of the organization. This may be an email address or an employee number. 
+
+The user data MAY also contain a `RoleName` that contains the user role within the organization.
+
+### 3.2 HTML based challenge
+
+After a session has been started by the node, the client application MUST direct the user to a web page that is hosted by the node.
+The node MUST make sure that the web page is linked to the session and that the give user data can not be altered.
+The web page MUST present the user with a contract as stated in [RFC002](./rfc002-authentication-token.md#5-login-contract) and show which user data will be exposed to the resource server. 
```

Comment:
The current UI says "Als u akkoord gaat worden deze gegevens gebruikt bij het opvragen van patient-gegevens bij andere zorginstellingen", so it's quite ambiguous what it will be used for (thus the UI isn't 100% according to this spec). Maybe it should be changed to "worden deze gegevens gedeeld met zorginstellingen"?